pandas/core/arrays/integer.py

```diff
@@ -154,7 +154,10 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             dtype = dtype.lower()
         if not issubclass(type(dtype), _IntegerDtype):
             try:
-                dtype = _dtypes[str(np.dtype(dtype))]
+                try:
```

Comment:
why are you changing this?

Reply:
if `dtype` is a class (not instantiated) then it's `type(dtype) == type` making this statement true, but then a `KeyError` gets raised because it can't find the `dtype` in the `_dtypes` dict. 

I know this isn't the best solution but I can't think of a better one. If you have a better solution then I would be happy to implement it.

Reply:
I think we can require that `dtype` is an instantiated object and not a class?, or where does this come up?

Reply:
I don't think requiring `dtype` to be instantiated is the correct behavior. The following test, which is based on the original issue, will fail with the error message: 
`ValueError: invalid dtype specified <class 'pandas.core.arrays.integer.Int64Dtype'>`

```python
@pytest.mark.parametrize('engine', ['c', 'python'])
def test_EA_types(engine):
    df = pd.DataFrame({'Int': pd.Series([1, 2, 3], dtype='Int64'),
                       'A': [1, 2, 1]})
    data = df.to_csv(index=False)
    result = pd.read_csv(StringIO(data), dtype={'Int': Int64Dtype},
                         engine=engine)
```

I guess we could insatiate the `dtype` somewhere further up in the call stack, but to me this seems like the correct place to do it. This is where we turn `str` objects into the correct `dtype`, so it makes sense that we handle class references here as well.

Reply:
this should not be changed